zfit/minimizers/fitresult.py

```diff
@@ -239,6 +242,50 @@ def _error(self, params, method, sigma):
                 raise KeyError("The following method is not a valid, implemented method: {}".format(method))
         return method(result=self, params=params, sigma=sigma)
 
+    def covariance(self, params: ParamsTypeOpt = None, as_dict: bool = False):
+        """Calculate the covariance matrix for `params`.
+
+            Args:
+                params (list(:py:class:`~zfit.Parameter`)): The parameters to calculate
+                    the covariance matrix. If `params` is `None`, use all *floating* parameters.
+                as_dict (bool): Default `False`. If `True` then returns a dictionnary.
+
+            Returns:
+                2D `numpy.array` of shape (N, N);
+                `dict`(param1, param2) -> covariance if `as_dict == True`.
+        """
+        params = self._input_check_params(params)
+        covariance_dict = self.minimizer._minuit_minimizer.covariance
```

Comment:
We need to be independent of minuit for sure. So for the moment, we can use the minuit covariance, but a standalone function should be there at some point (using _any_ library, we can use pure np for the calulations as well)

Can you put it into a try-except block catching the `AttributeError` and saying that only minuit covariance is currently there?

Reply:
Ok I agree but `MinuitMinimizer`doesn't have a covariance method. What's your idea of covariance implementation with other minimiser?

Reply:
Something similar to minuit for example. If I understand that correctly, it's an inversion of the error matrix, removal of the line and column that contains the parameter and a re-inversion.

Getting the error matrix, by hessian or by minos-like profiling, is another story, but also doable

Reply:
Yeah it's something like this, like the inverse of the hessian matrix of the log-likelihood.